src/engine/Server.cpp

```diff
@@ -1036,13 +1038,12 @@ Awaitable<void> Server::processQueryOrUpdate(
         LOG(ERROR) << metadata.value().query_ << std::endl;
       }
     }
-    std::string operationStr = [&operation] {
+    const std::string operationStr = [&operation] {
       if constexpr (std::is_same_v<Operation, Query>) {
         return operation.query_;
-      } else if constexpr (std::is_same_v<Operation, Update>) {
-        return operation.update_;
       } else {
-        AD_FAIL();
+        static_assert(std::is_same_v<Operation, Update>);
+        return operation.update_;
```

Comment:
Do you need to copy the (possibly large) string here?
Either return and store a `const&` from the lambda, or at least move the string into the following functions (if composeErrorResponseJson requires a string by value).

Reply:
It seems that the return also has to be specified explicitly in this case ([Experiment in Godbolt](https://godbolt.org/z/E14b6aW4P)).